src/test/components/TooltipMarker.test.js

```diff
@@ -955,4 +956,98 @@ describe('TooltipMarker', function () {
 
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  it('shows the source thread for markers from a merged thread', function () {
+    // We construct a profile that has 2 threads from 2 different tabs.
+    const { profile } = getProfileFromTextSamples(`A`, `A`);
+    profile.threads[0] = {
+      ...profile.threads[0],
+      name: 'GeckoMain',
+      processName: 'Isolated Web Content',
+      'eTLD+1': 'https://mozilla.org',
+    };
+    profile.threads[1] = {
+      ...profile.threads[1],
+      name: 'GeckoMain',
+      processName: 'Isolated Web Content',
+      'eTLD+1': 'https://letsencrypt.org',
+    };
+
+    const innerWindowID1 = 1;
+    const tabID1 = 10;
+    const innerWindowID2 = 2;
+    const tabID2 = 11;
+    profile.pages = [
```

Comment:
Ah yeah, but this tool doesn't let me specify the URL, and I thought it would be good to be consistent... I'll keep it this way then, it's a small array and I believe it's easier to follow this way.